### PR TITLE
Revamps project to recent version of Brave and refactors senders

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Before, I think we accidentally conflated zipkin reporter's senders with
instrumentation. When we go to instrument okhttp, for example, that will
different than the sender (what sends spans to zipkin).

This doesn't pass tests, so could use a hand.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
@cschneider @reta Not sure if this looks obvious to you.. The itests fail with what looks like an unresolved variable

```
Caused by: java.io.IOException: Could not download [file:/Users/acole/.m2/repository/io/zipkin/reporter2/zipkin-sender-urlconnection/2.7.7/zipkin-sender-urlconnection-2.7.7.jar${zipkin.reporter.version}]
	at org.apache.karaf.features.internal.download.impl.SimpleDownloadTask.download(SimpleDownloadTask.java:90)[8:org.apache.karaf.features.core:4.0.7]
	at org.apache.karaf.features.internal.download.impl.AbstractRetryableDownloadTask.run(AbstractRetryableDownloadTask.java:58)[8:org.apache.karaf.features.core:4.0.7]
	... 7 more
Caused by: java.io.FileNotFoundException: /Users/acole/.m2/repository/io/zipkin/reporter2/zipkin-sender-urlconnection/2.7.7/zipkin-sender-urlconnection-2.7.7.jar${zipkin.reporter.version} (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)[:1.8.0_162]
	at java.io.FileInputStream.open(FileInputStream.java:195)[:1.8.0_162]
	at java.io.FileInputStream.<init>(FileInputStream.java:138)[:1.8.0_162]
	at java.io.FileInputStream.<init>(FileInputStream.java:93)[:1.8.0_162]
	at sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:90)[:1.8.0_162]
	at sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:188)[:1.8.0_162]
	at java.net.URL.openStream(URL.java:1045)[:1.8.0_162]
	at org.apache.karaf.features.internal.download.impl.SimpleDownloadTask.download(SimpleDownloadTask.java:76)[8:org.apache.karaf.features.core:4.0.7]
```

For the life of me, I can't figure out why...

File: features/src/main/resources/features.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+-->
+<features name="brave" xmlns="http://karaf.apache.org/xmlns/features/v1.2.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://karaf.apache.org/xmlns/features/v1.2.0 http://karaf.apache.org/xmlns/features/v1.2.0">
+
+	<feature name="brave">
+		<feature>scr</feature>
+		<bundle dependency="true">mvn:io.zipkin.zipkin2/zipkin/${zipkin.version}</bundle>
+		<bundle dependency="true">mvn:io.zipkin.reporter2/zipkin-reporter/${zipkin.reporter.version}</bundle>
```
Comment (codefromthecrypt) [inline features/src/main/resources/features.xml]: ex no idea if this is what is resulting in the itests error. If it is no idea why :)

File: exporter/src/main/java/io/zipkin/brave/exporter/sender/urlconnection/URLConnectionSenderExporter.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  @Activate
+  public void activate(Config config, BundleContext context, Map<String, String> properties) {
+    sender = URLConnectionSender.newBuilder()
+        .endpoint(config.endpoint())
+        .compressionEnabled(config.compressionEnabled())
+        .connectTimeout(config.connectTimeout())
+        .messageMaxBytes(config.messageMaxBytes())
+        .build();
+    reg = context.registerService(Sender.class, sender, new Hashtable<String, String>(properties));
```
Comment (codefromthecrypt) [inline exporter/src/main/java/io/zipkin/brave/exporter/sender/urlconnection/URLConnectionSenderExporter.java:48]: the build complains about Hashtable. does registerService expect a synchronized input? If not I'd like to switch this to LinkedHashMap to quiet the warnings

**Comment by codefromthecrypt** [comment]
@jbonofre if you are bored or interested.. this hasn't been released yet in any way.. mostly pay attention to green as it is restructured to fit our builds. This ideally ends up with tracing exported in karaf

File: itests/pom.xml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  <parent>
+    <groupId>io.zipkin.brave.karaf</groupId>
+    <artifactId>brave-karaf-parent</artifactId>
+    <version>0.1.0-SNAPSHOT</version>
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+  <artifactId>brave-itests</artifactId>
+
+  <properties>
+    <pax-exam.version>4.12.0</pax-exam.version>
```
Comment (codefromthecrypt) [inline itests/pom.xml:30]: this inherits osgi from parent which is 6.0.0. not sure if this is good or bad

File: features/src/main/resources/features.xml (hunk 1 above)
Comment (cschneider) [reply features/src/main/resources/features.xml]: There is a typo. The property in the parent is named zipkin-reporter.version. Once this is fixed there is a unresolved requirement. I will dig into it.

**Comment by codefromthecrypt** [comment]
yep looks like a typo! good eyes. trying build locally

**Comment by codefromthecrypt** [comment]
gets farther... to this. I'll let you give a try as you might be faster
fixing :)

org.osgi.service.resolver.ResolutionException: Unable to resolve root:
missing requirement [root] osgi.identity; osgi.identity=brave;
type=karaf.feature; version="[0,0.0.0]";
filter:="(&(osgi.identity=brave)(type=karaf.feature)(version>=0.0.0)(version<=0.0.0))"
[caused by: Unable to resolve brave/0.0.0: missing requirement
[brave/0.0.0] osgi.identity; osgi.identity=brave-exporter;
type=osgi.bundle; version="[0.1.0.201808010758,0.1.0.201808010758]";
resolution:=mandatory [caused by: Unable to resolve
brave-exporter/0.1.0.201808010758: missing requirement
[brave-exporter/0.1.0.201808010758] osgi.extender;
filter:="(&(osgi.extender=osgi.component)(version>=1.4.0)(!(version>=2.0.0)))"]]

**Comment by reta** [comment]
@adriancole 

Not sure why tests are failing, but looking into `feature.xml` I see 2 variants: `${zipkin.reporter.version}` and `${zipkin-reporter.version}`, could it be that?

```
<feature name="brave">
    ...
    <bundle dependency="true">mvn:io.zipkin.reporter2/zipkin-reporter/${zipkin-reporter.version}</bundle>
    <bundle>mvn:io.zipkin.reporter2/zipkin-sender-urlconnection/${zipkin-reporter.version}</bundle>
    ...
</feature>
	
<feature name="brave-sender-kafka">
        ...
	<bundle>mvn:io.zipkin.reporter2/zipkin-sender-kafka11/${zipkin.reporter.version}</bundle>
        ...
</feature>

<feature name="brave-sender-okhttp`">
        ...
     	<bundle>mvn:io.zipkin.reporter2/zipkin-sender-okhttp3/${zipkin.reporter.version}</bundle>
        ...
</feature>
```

**Comment by codefromthecrypt** [comment]
WOW thank you so much @cschneider !

**Pull request merged**